dual_net.py

```diff
@@ -305,30 +293,31 @@ def export_model(working_dir, model_path):
         tf.gfile.Copy(filename, destination_path)
 
 
-def train(working_dir, tf_records, steps=None, **hparams):
-    estimator = get_estimator(working_dir, **hparams)
+def train(working_dir, tf_records, steps=None):
+    estimator = get_estimator(working_dir)
 
     def input_fn():
-        return preprocessing.get_input_tensors(TRAIN_BATCH_SIZE, tf_records,
-                                               filter_amount=1.0)
+        return preprocessing.get_input_tensors(
+            FLAGS.train_batch_size, tf_records, filter_amount=1.0,
+            shuffle_buffer_size=FLAGS.shuffle_buffer_size)
 
     update_ratio_hook = UpdateRatioSessionHook(working_dir)
     step_counter_hook = EchoStepCounterHook(output_dir=working_dir)
     if steps is None:
-        steps = EXAMPLES_PER_GENERATION // TRAIN_BATCH_SIZE
+        steps = EXAMPLES_PER_GENERATION // FLAGS.train_batch_size
     estimator.train(input_fn, hooks=[
                         update_ratio_hook, step_counter_hook], steps=steps)
 
 
-def validate(working_dir, tf_records, checkpoint_name=None, **hparams):
-    estimator = get_estimator(working_dir, **hparams)
+def validate(working_dir, tf_records, checkpoint_name=None, validate_name=None):
```

Comment:
apparently the **kwargs was swallowing the fact that we were passing in invalid kwarg names.

Reply:
which invalid kwarg names were we passing?

Reply:
we were passing name=validate_name in main.py